src/components/engineer/Engineer.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState, createContext } from "react";
+import Plane from "../plane/Plane";
+import EngineerRows from "./EngineerRows";
+import { EngineerFieldsCheck } from "../../helper";
+export const EngineerContext = createContext([]);
+const Engineer = () => {
+  const [engineerrows, setEngineerrows] = useState([]);
+  const [name, setName] = useState("");
+  const [namearr, setNamearr] = useState([]);
+  const [age, setAge] = useState("");
+  const [exp, setExp] = useState("");
+  return (
+    <div>
+      <form name="engineers">
+        <h1>Engineer Description</h1>
+        <label>Name</label>
+        <input
+          value={name}
+          onChange={(en) => setName(en.target.value)}
+          minLength="2"
+          required
+        />
+
+        <br />
+        <label>Age</label>
+        <input
+          value={age}
+          onChange={(e) => {
+            if (!isNaN(e.target.value)) {
+              setAge(e.target.value);
+            }
+          }}
+          type="numbers"
+          required
+        />
+        <br />
+
+        <label>Experience</label>
+        <input
+          value={exp}
+          onChange={(e) => {
+            if (!isNaN(e.target.value)) {
+              setExp(e.target.value);
+            }
+          }}
+          type="numbers"
+          required
+        />
+        <br />
+        <button
+          type="button"
+          onClick={() => {
+            if (EngineerFieldsCheck(name, age, exp)) {
+              console.log(EngineerFieldsCheck(name, age, exp));
+              setEngineerrows((old) => [
```

Comment:
set object instead of row components.

create a function and use it here instead of making inline function

Reply:
eg, 
```
const addEngineer = ()=>{...}
...
onClick={addEngineer}

```